### PR TITLE
Display dartdoc status on the versions page.

### LINKED ISSUE #993: Show dartdoc generation failure log links in the package version table
If dartdoc generation fails for a package version, replace the current docs icon button with the work 'Failed' and a hyperlink the logfile for that.

### PR DESCRIPTION
**Pull request by isoos**
Closes #993 

https://dartlang-pub-dev.appspot.com/packages/url_launcher#-versions-tab-
https://dartlang-pub-dev.appspot.com/packages/url_launcher/versions

Unfortunately the failed ones are not yet exposed due to a bug in dartdoc_runner:
https://dartlang-pub-dev.appspot.com/packages/audio_recorder/versions

### CODE REVIEW

**Comment by isoos** [comment]
I have a few code cleanup tasks in the `frontend/templates` and `frontend/handlers` code that is a remnant of the v2 style migration. I'll do a follow-up on those after this got merged.

**Comment by isoos** [comment]
It looks like the failure use case is now working too (with the directory creation fix).

File: app/views/pkg/versions/version_row.mustache [hunk 1]
```diff
@@ -6,13 +6,17 @@
   <td><strong><a href="/packages/{{package}}/versions/{{version}}">{{version}}</a></strong></td>
   <td>{{short_created}}</td>
   <td class="documentation">
-    <a href="{{& documentation_url}}"
+    <a class="version-table-icon" href="{{& documentation_url}}"
        title="Go to the documentation of {{package}} {{version}}">
       <img src="{{& icons.documentation}}" alt="Go to the documentation of {{package}} {{version}}" />
     </a>
+    <span class="version-table-status">
+      {{#dartdoc_ok}}(OK){{/dartdoc_ok}}
```
Comment (jakobr-google) [inline app/views/pkg/versions/version_row.mustache]: I'm not sure we want to explicitly show an OK status - that should be the expectation. I'd prefer just showing the failed status, if applicable.

File: app/views/pkg/versions/version_row.mustache [hunk 2]
```diff
@@ -6,13 +6,17 @@
   <td><strong><a href="/packages/{{package}}/versions/{{version}}">{{version}}</a></strong></td>
   <td>{{short_created}}</td>
   <td class="documentation">
-    <a href="{{& documentation_url}}"
+    <a class="version-table-icon" href="{{& documentation_url}}"
```
Comment (jakobr-google) [inline app/views/pkg/versions/version_row.mustache]: Maybe only show the documentation link if `dartdoc_ok`?

Comment (isoos) [reply app/views/pkg/versions/version_row.mustache]: SGTM, will update.

**Comment by isoos** [comment]
PTAL

**Comment by isoos** [comment]
An example for the failed links:
https://dartlang-pub-dev.appspot.com/packages/dartdoc/versions

**Pull request merged**